oak_functions/loader/src/server.rs

```diff
@@ -1291,4 +1386,17 @@ mod tests {
             .expect("No extension for channel handle.");
         uwabi_extensions.remove(pos_of_extension)
     }
+
+    // Helper function for testing to write a buffer of i32s in Wasm memory.
+    fn write_i32_buffer_to_wasm_memory(
+        wasm_state: &mut WasmState,
+        i32_buffer: &Vec<i32>,
+        dest: AbiPointer,
+    ) -> Result<(), ChannelStatus> {
+        let mut bytes_buffer: Vec<u8> = Vec::new();
+        for i in i32_buffer {
+            bytes_buffer.extend_from_slice(&i.to_le_bytes());
+        }
```

Comment:
If you want, you could also extract the core functionality of those two functions (converting bytes to i32, and vice versa), and then have a property based test to show that they are one the inverse of the other (as long as the buffer is correctly sized, of course).